.drone.yml

```diff
@@ -1,129 +1,200 @@
 ---
-pipeline:
-  build:
-    privileged: true
-    image: rancher/dapper:1.11.2
-    volumes:
-    - /var/run/docker.sock:/var/run/docker.sock
-    commands:
-    - dapper ci
-    when:
-      event: [ push, pull_request, tag ]
+kind: pipeline
+name: default
 
-  stage-binaries:
-    image: rancher/dapper:1.11.2
-    commands:
-    - cp -r ./bin/* ./package/
-    when:
-      branch: [master, release/*]
-      event: [push,tag]
+platform:
+  os: linux
+  arch: amd64
 
-  docker-publish-master:
-    image: plugins/docker
+steps:
+- name: build
+  pull: default
+  image: rancher/dapper:v0.4.1
+  commands:
+  - dapper ci
+  privileged: true
+  volumes:
+  - name: docker
+    path: /var/run/docker.sock
+  when:
+    event:
+    - push
+    - pull_request
+    - tag
+
+- name: stage-binaries
+  pull: default
+  image: rancher/dapper:v0.4.1
+  commands:
+  - "cp -r ./bin/* ./package/"
+  when:
+    event:
+    - push
+    - tag
+
+- name: docker-publish-master
+  pull: default
+  image: plugins/docker
+  settings:
+    build_args:
+    - VERSION=master
+    context: package/
+    custom_dns: 1.1.1.1
     dockerfile: package/Dockerfile
+    password:
+      from_secret: docker_password
     repo: rancher/rancher
-    context: package/
     tag: master
-    secrets: [docker_username, docker_password]
-    environment:
-    -  PLUGIN_CUSTOM_DNS=1.1.1.1
+    username:
+      from_secret: docker_username
+  when:
+    branch:
+    - master
+    event:
+    - push
+
+- name: docker-publish-master-agent
+  pull: default
+  image: plugins/docker
+  settings:
     build_args:
     - VERSION=master
-    when:
-      branch: master
-      event: push
-
-  docker-publish-master-agent:
-    image: plugins/docker
+    context: package/
+    custom_dns: 1.1.1.1
     dockerfile: package/Dockerfile.agent
+    password:
+      from_secret: docker_password
     repo: rancher/rancher-agent
-    context: package/
     tag: master
-    secrets: [docker_username, docker_password]
-    environment:
-    -  PLUGIN_CUSTOM_DNS=1.1.1.1
-    build_args:
-    - VERSION=master
-    when:
-      branch: master
-      event: push
+    username:
+      from_secret: docker_username
+  when:
+    branch:
+    - master
+    event:
+    - push
 
-  docker-publish:
-    image: plugins/docker
+- name: docker-publish
+  pull: default
+  image: plugins/docker
+  settings:
+    build_args:
+    - "VERSION=${DRONE_TAG}"
+    context: package/
+    custom_dns: 1.1.1.1
     dockerfile: package/Dockerfile
+    password:
+      from_secret: docker_password
     repo: rancher/rancher
-    context: package/
-    tag: ${DRONE_TAG}
-    secrets: [docker_username, docker_password]
-    environment:
-    -  PLUGIN_CUSTOM_DNS=1.1.1.1
-    build_args:
-    - VERSION=${DRONE_TAG}
-    when:
-      branch: [master, release/*]
-      event: tag
+    tag: "${DRONE_TAG}"
+    username:
+      from_secret: docker_username
+  when:
+    event:
+    - tag
+    ref:
+    - refs/head/master
```

Comment:
Is it necessary to have the ref: block here.  At least with drone 0.8 Tags only have the proper ref if the commit tagged it he current head of the branch.